tests/variable_creation_test.py

```diff
@@ -570,6 +570,18 @@ def test_arange_with_variables_requires_scalar():
         sc.arange('x', sc.scalar(1), sc.array(dims=['x'], values=[1, 2]))
 
 
+def test_arange_with_variables_does_not_allow_variances():
+    start = sc.scalar(1)
+    stop = sc.scalar(4)
+    step = sc.scalar(1)
+    with pytest.raises(sc.VariancesError):
+        sc.arange('x', start, stop, sc.scalar(1.0, 0.1))
+    with pytest.raises(sc.VariancesError):
+        sc.arange('x', start, sc.scalar(4.0, 0.1), step)
+    with pytest.raises(sc.VariancesError):
+        sc.arange('x', sc.scalar(1.0, 0.1), stop, step)
```

Comment:
Don't rely on non-keyword args?

Reply:
Why? I don't think we should ever change the order or make them keyword only.

Reply:
I would  argue that we *should* make `variance` keyword only? As it is now it is very unclear.

Reply:
Oh, I thought you were talking about `arange` and not `scalar`. I can change this now to make the PR for `scalar` easier.

Reply:
Yes that was my point.